src/api/resources/project.py

```diff
@@ -381,6 +383,34 @@ def create_project(self, request):
         transaction.savepoint_commit(save_point_id)
         return Response(project_serializer.data)
 
+    @action(
+        detail=False,
+        methods=["post"],
+        permission_classes=[ProjectAuthenticatedUserPermission],
+    )
+    def create_demo(self, request):
+        if Project.objects.filter(created_by=request.user.profile, is_demo=True).exists():
+            raise exceptions.ValidationError(
+                detail="You have already created a demo project. Only one demo project is allowed per user."
+            )
+
+        unique_id = ""
+        tries = 0
+        while True:
+            unique_id = "".join(random.choices(string.ascii_letters + string.digits, k=4))
+            project_name = f"Demo - {unique_id}"
```

Comment:
Hmmm... The mockups don't specify this, but it's true making something unique is desirable. What about using the owner's name?